### PR TITLE
cNetwork: Added UDP API.

### PR DESCRIPTION
**Pull request by madmaxoft**
This PR adds UDP functionality to the cNetwork API, including UDP datagram broadcast.

### CODE REVIEW

File: src/OSSupport/UDPEndpointImpl.cpp [hunk 1]
```diff
@@ -0,0 +1,608 @@
...
+
+
+
+bool cUDPEndpointImpl::Send(const AString & a_Payload, const AString & a_Host, UInt16 a_Port)
+{
+	// If a_Host is an IP address, send the data directly:
+	sockaddr_storage sa;
+	socklen_t salen = static_cast<int>(sizeof(sa));
+	memset(&sa, 0, sizeof(sa));
+	if (evutil_parse_sockaddr_port(a_Host.c_str(), reinterpret_cast<sockaddr *>(&sa), &salen) != 0)
```
Comment (worktycho) [inline src/OSSupport/UDPEndpointImpl.cpp]: I think you forgot to include event2/util.h

Comment (madmaxoft) [inline src/OSSupport/UDPEndpointImpl.cpp]: No, it's there, it's just a type mismatch - int vs unsigned vs size_t. These APIs are a royal PITA to work with multi-platformly.
Weird thing is, the compilation had already succeeded in a previous build: https://travis-ci.org/mc-server/MCServer/builds/51497191

**Comment by madmaxoft** [comment]
I think this is now ready for review and merge.

**Comment by NiLSPACE** [comment]
Looks good to me.

**Pull request merged**